app/config.py

```diff
@@ -103,48 +73,534 @@
-        in from defaults. If a non-empty prefix string is supplied,
-        and any environment variables exist beginning with that
-        prefix, add those values to the dictionary overwriting
-        anything in sdict or default.
-        """
-        super(Map, self).__init__(dict(sdict))
-        self.prefix = ("" if prefix == "" else prefix + "_").upper()
-
-        # If any values are missing, copy them from defaults.
-        # Turn all subdictionaries into Maps as well.
-        for key, val in defaults.items():
-            if isinstance(val, dict):
-                existing = self.get(key, {})
-                if existing is None:
-                    existing = {}
-                elif not isinstance(existing, dict):
+# Nested map of configuration keys which are configurable in the admin
+# interface. The accompanying value will only be used in the migration
+# which writes that value to the database, and in the tests.
+configurable_defaults = {
+    "site": {
+        "type": "map",
+        "value": {
+            "name": {
+                "type": "string",
+                "doc": _l("Name of the site."),
+                "value": "Throat",
+            },
+            "lema": {
+                "type": "string",
+                "doc": _l("Lema shown in the page's title."),
+                "value": "Throat: Open discussion ;D",
+            },
+            "copyright": {
+                "type": "string",
+                "doc": _l("Copyright line shown in the footer."),
+                "value": "Umbrella Corp",
+            },
+            "placeholder_account": {
+                "type": "string",
+                "doc": _l(
+                    "Name shown when a sub is owned by a deleted account or abandoned."
+                ),
+                "value": "Site",
+            },
+            "enable_security_question": {
+                "type": "bool",
+                "doc": _l(
+                    "If True, enables setting security questions on the admin page. Users will be "
+                    "asked to answer one of these security questions before registering."
+                ),
+                "value": False,
+            },
+            "allow_uploads": {
+                "type": "bool",
+                "doc": _l(
+                    "Allow everybody to upload files (by default it's only admins and users "
+                    "authorized through a metadata key)."
+                ),
+                "value": False,
+            },
+            "allow_video_uploads": {
+                "type": "bool",
+                "doc": _l(
+                    "For those who are allowed to upload files, allow video uploads (.mp4 and .webm) as well."
+                ),
+                "value": True,
+            },
+            "upload_max_size": {
+                "type": "int",
+                "doc": _l("Maximum size of an uploaded file, in bytes."),
+                "value": 16777216,
+            },
```

Comment:
I'd move this back to the config.yaml since it will be aliased with `max_content_length`, which is an internal flask config that probably requires a restart to take effect.

Reply:
Done.